File: gate-domain/src/main/java/gate/domain/session/AgentConfig.java
package gate.domain.session;

import java.util.List;

/**
 * A reusable agent configuration (≈ multica Agent, but no Squad/Skill/Runtime) — 执行文档-后端-web §5.2.
 *
 * @param id             e.g. "claude-sonnet-default"
 * @param name           display name
 * @param cli            OPENCODE | CLAUDE
 * @param providerId     optional provider table id for a future API-backed runtime; local CLI
 *                       sessions leave this unset so the CLI owns provider selection
 * @param model          optional model override; local CLI sessions leave this unset to use the
 *                       CLI's own default configuration
 * @param systemPrompt   optional extra system prompt
 * @param extraFlags     passthrough CLI flags
 * @param description    optional description
 * @param injectContext  会话启动时把项目信息与工单信息作为系统提示词注入；缺省开启
 */
public record AgentConfig(
        String id,
        String name,
        AgentCli cli,
        String providerId,
        String model,
        String systemPrompt,
        List<String> extraFlags,
        String description,
        boolean injectContext) {

    /** Legacy shape (pre inject-context); injection defaults to on. */
    public AgentConfig(String id, String name, AgentCli cli, String providerId, String model,
                       String systemPrompt, List<String> extraFlags, String description) {
        this(id, name, cli, providerId, model, systemPrompt, extraFlags, description, true);
    }

    public AgentConfig {
        if (id == null || id.isBlank()) {
            throw new IllegalArgumentException("id must not be blank");
        }
        if (name == null || name.isBlank()) {
            throw new IllegalArgumentException("name must not be blank");
        }
        if (cli == null) {
            throw new IllegalArgumentException("cli must not be null");
        }
        providerId = nullable(providerId);
        model = nullable(model);
        extraFlags = extraFlags == null ? List.of() : List.copyOf(extraFlags);
    }

    private static String nullable(String value) {
        return value == null || value.isBlank() ? null : value;
    }
}

File: gate-web-ui/src/types/agentConfig.ts
/**
 * AgentConfig — 对齐后端 gate-domain.session.AgentConfig / AgentCli (后端文档 §5.2).
 */
export type AgentCli = 'OPENCODE' | 'CLAUDE';

export interface AgentConfig {
  id: string;
  name: string;
  cli: AgentCli;
  /** API runtime 的可选关联；本机 CLI agent 通常为空。 */
  providerId?: string | null;
  /** 可选模型覆盖；为空时由 CLI 使用自身默认模型。 */
  model?: string | null;
  /** 可空; 额外 system prompt. */
  systemPrompt?: string | null;
  /** JSON array of strings; 透传 CLI flag. */
  extraFlags?: string[] | null;
  description?: string | null;
}

/** 模型供应商 (后端 ProviderRepository record 投影). */
export interface Provider {
  id: string;
  name: string;
  /** 旗下模型 id 列表. */
  models: string[];
}
